packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -534,14 +548,18 @@ export async function getPastExperimentsByDatasource(
     }
   );
 
-  return experiments.map((exp) => ({
+  const experimentsUserCanAccess = experiments.filter((exp) =>
+    hasReadAccess(context.readAccessFilter, exp.project)
```

Comment:
This won't work.  In the above db query, we're only returning the `id` and `trackingKey` field, not the `project` field so this will always be undefined.